cobc/ChangeLog

```diff
@@ -1,4 +1,14 @@
 
+2022-12-05  Nicolas Berthier <nicolas.berthier@ocamlpro.com>
+
+	* cobc.h (struct cb_replace_src, struct list_replace): adapt internal
+	  structures for strict partial text replacement
+	* cobc.c (print_replace_text, print_replace_main),
+	  pplex.l (set_print_replace_list, ppecho),
+	  ppparse.y (ppp_replace_src, ppp_replace_list_add): handle strict
+	  partial text replacement
+	* cobc.c: replace some incorrect '%d' formats indicators with '%lu'
```

Comment:
the format fixing could be checked in directly to minimize the diff, if you want to; this also applies to moving the conversion table in the "if" from https://github.com/OCamlPro/gnucobol/pull/71/files#diff-e2d9f5867efd7382ed31747ef1d5c8c495c20ea34c4e2b58abc7b019e7407f13 - feel free to commit one or both directly